### PR TITLE
UI test kangquan

### PR DESCRIPTION
**Pull request by Kang-Quan**
UI Test: 
Create UI test for registering an account:
click on register -> Register user by filling in the inputs -> creates a new db entry -> check whether they redirect to /login and says registered -> fill up log in details -> check for success message

2. Create UI testcase for unauthenticated adding to cart then checkout
add to cart (unauthenticated) -> go to cart page -> check rendering of cart page -> proceed to log in -> check rendering of cart page after log in

3. Create UI testing for checking cart is correct after removing and re-login:
log in -> Adding 2 item cart -> go to cart page -> user changes mind -> delete 1 from cart -> check rendering -> log out -> log in back -> go to cart page -> 1 item should still be in the cart page

### CODE REVIEW

File: playwright.config.cjs [hunk 1]
```diff
@@ -26,6 +26,7 @@ module.exports = defineConfig({
   /* Reporter to use. See https://playwright.dev/docs/test-reporters */
   reporter: 'html',
   /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
+  timeout: 60000,
```
Comment (Aishwarya-Hariharan-Iyer) [inline playwright.config.cjs:29]: Not sure if this increases/decreases from the previous setting. If the latter, we need to check if existing tests fail.

File: tests/authenticatedAddingAndRemoving.spec.cjs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	
+    //ensure correct rendering of items in cart
+	await expect(page.getByText(productName2, { exact: true })).toBeVisible();
+	await expect(
+		page.getByText("this is the best book2", { exact: true })
+	).toBeVisible();
+	await expect(page.getByText("price : 19")).toBeVisible();
+});
+
+function generateRandomProductName() {
```
Comment (Aishwarya-Hariharan-Iyer) [inline tests/authenticatedAddingAndRemoving.spec.cjs:146]: Might be better to separate helper functions into their own file? Just my thoughts :))

File: tests/register.spec.cjs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+test.beforeAll(async () => {
+	// @ts-ignore
+	client = new MongoClient(process.env.MONGO_URL);
+	await client.connect();
+	db = client.db("ecommerce-test");
+});
+
+let email = generateRandomEmail();
+
+function generateRandomEmail() {
```
Comment (Aishwarya-Hariharan-Iyer) [inline tests/register.spec.cjs:19]: Same - might help avoid redundancy too

File: tests/unauthenticatedCheckout.spec.cjs [hunk 4]
```diff
@@ -0,0 +1,158 @@
+import { test, expect } from "@playwright/test";
+const { MongoClient, ObjectId } = require("mongodb");
+import dotenv from "dotenv";
+
+dotenv.config();
+
+let db;
+let client;
+
+let email = generateRandomEmail();
```
Comment (Aishwarya-Hariharan-Iyer) [inline tests/unauthenticatedCheckout.spec.cjs:10]: same

**Review by Aishwarya-Hariharan-Iyer — APPROVED** [review]
LGTM! A few cosmetic/stylistic changes suggested :))

File: playwright.config.cjs (hunk 1 above)
Comment (Kang-Quan) [reply playwright.config.cjs:29]: This is an increase from previous setting as I have one test that takes a longer path. Default setting is 30 seconds I doubled it to be on the safe side. documentations here: https://playwright.dev/docs/test-timeouts

File: tests/authenticatedAddingAndRemoving.spec.cjs (hunk 2 above)
Comment (Kang-Quan) [reply tests/authenticatedAddingAndRemoving.spec.cjs:146]: I agree with you helper function should have their own file. But for this function it is being used once thus leaving it in the same file.

File: tests/register.spec.cjs (hunk 3 above)
Comment (Kang-Quan) [reply tests/register.spec.cjs:19]: I agree with you. Same reason as above should more files require this random email function, I will abstract it into a helper function folder in subsequent commits.

File: tests/unauthenticatedCheckout.spec.cjs (hunk 4 above)
Comment (Kang-Quan) [reply tests/unauthenticatedCheckout.spec.cjs:10]: same reason as above.

**Pull request merged**